android/app/src/main/kotlin/coop/polypoly/polypod/polyNav/PolyNav.kt

```diff
@@ -65,16 +69,18 @@ open class PolyNav(
                     )
             }
         }
-        contentResolver?.openInputStream(importedUrl).use { inputStream ->
-            if (inputStream == null) {
-                throw Error("File copy error")
+        coroutineScope.async {
+            contentResolver?.openInputStream(importedUrl).use { inputStream ->
+                if (inputStream == null) {
+                    throw Error("File import error")
+                }
+                val newId = UUID.randomUUID().toString()
+                val fs = Preferences.getFileSystem(context).toMutableMap()
+                fs[fsPrefix + newId] = fileName
+                Preferences.setFileSystem(context, fs)
+                ZipTools.unzipAndEncrypt(inputStream, context, newId)
             }
-            val newId = UUID.randomUUID().toString()
-            ZipTools.unzipAndEncrypt(inputStream, context, newId)
-            val fs = Preferences.getFileSystem(context).toMutableMap()
-            fs[fsPrefix + newId] = fileName
-            Preferences.setFileSystem(context, fs)
-        }
+        }.await()
```

Comment:
I thought `withContext` was the more common/idiomatic way of doing just this. But it should work, so good enough for me :P